guides/v2.3/extension-dev-guide/message-queues/refresh-config.md

```diff
@@ -0,0 +1,35 @@
+---
+group: php-developer-guide
+title: Handling outdated in-memory object states
+functional_areas:
+  - Configuration
+  - System
+  - Setup
+---
+
+When Magento is launched, the store's configuration is loaded and copied into memory. Magento uses that copy in memory for queue message transactions. When the store's configuration is updated in the admin, the copy in memory does not automatically refresh resulting in an outdated in-memory object state.
+
+To ensure the copy in memory is re-instantiated after an update, use the poison pill interface available in the MessageQueue module. Before a new message is processed, the poison pill interface compares the version of the in-memory copy to the latest in the database. If the versions are different, the in-memory copy is destroyed and a new copy is created when the next cron:run job executes.
```

Comment:
" is processed, the poison pill interface compares the version" - the implementation of PoisonPillCompareInterface compares the version